build-coordinator/src/main/java/org/jboss/pnc/core/builder/coordinator/bpm/BpmBuildScheduler.java

```diff
@@ -122,103 +113,51 @@ ProcessInstance startProcess(BuildTask buildTask, Integer buildTaskSetId) throws
+
+        return objectMapper.writeValueAsString(params);
     }
 
-    void fillBuildRequest(Map<String, Object> parameters, BuildTask buildTask, Integer buildTaskSetId)
-            throws JsonProcessingException {
-        Map<String, Object> buildRequest = new HashMap<>();
-        buildRequest.put("buildTaskId", buildTask.getId());
-        buildRequest.put("buildTaskSetId", buildTaskSetId);
-        buildRequest.put("buildConfiguration",buildTask.getBuildConfiguration().getId());
-        buildRequest.put("buildConfigurationRevision",
-                Optional.of(buildTask)
-                        .map(BuildTask::getBuildConfigurationAudited)
-                        .map(BuildConfigurationAudited::getIdRev)
-                        .map(IdRev::getRev)
-                        .orElse(null));
-        buildRequest.put("buildRecordSetIdsCSV", StringUtils.toCVS(buildTask.getBuildRecordSetIds()));
-        buildRequest.put("buildConfigSetRecordId", buildTask.getBuildConfigSetRecordId());
-        buildRequest.put("buildContentId", ContentIdentityManager.getBuildContentId(buildTask.getBuildConfiguration()));
-        buildRequest.put("submitTimeMillis", buildTask.getSubmitTime().getTime());
-        buildRequest.put("pncUsername", Optional.of(buildTask)
-                .map(BuildTask::getUser)
-                .map(User::getUsername)
-                .orElse(null));
-        buildRequest.put("pncUserLoginToken", Optional.of(buildTask)
-                .map(BuildTask::getUser)
-                .map(User::getLoginToken)
-                .filter(token -> !"no-token".equals(token))
-                .orElse(null));
-        buildRequest.put("pncUserEmail", Optional.of(buildTask)
-                .map(BuildTask::getUser)
-                .map(User::getEmail)
-                .orElse(null));
-
-        parameters.put("buildRequestJSON", objectMapper.writeValueAsString(buildRequest));
+    String getBuildExecutionConfiguration(BuildTask buildTask) {
+
+        BuildConfiguration buildConfiguration = buildTask.getBuildConfiguration();
+        String contentId = ContentIdentityManager.getBuildContentId(buildConfiguration.getName());
+
+        BuildExecutionConfiguration buildExecutionConfiguration = BuildExecutionConfiguration.build(
+                buildTask.getId(),
+                contentId,
+                buildTask.getUser().getId(),
+                buildConfiguration.getBuildScript(),
+                buildConfiguration.getName(),
+                buildConfiguration.getScmRepoURL(),
+                buildConfiguration.getScmRevision(),
+                buildConfiguration.getScmMirrorRepoURL(),
+                buildConfiguration.getScmMirrorRevision(),
+                buildConfiguration.getBuildEnvironment().getBuildType());
+
+        BuildExecutionConfigurationRest buildExecutionConfigurationREST = new BuildExecutionConfigurationRest(buildExecutionConfiguration);
+
+        return buildExecutionConfigurationREST.toString();
```

Comment:
Proper serialization is used in `BuildExecutionConfigurationREST.toString()` method. 
Are you suggesting to not use `toString` for this purpose and rather say `toJson`?